src/main/java/org/openstreetmap/atlas/checks/validation/linear/lines/WaterWayCheck.java

```diff
@@ -0,0 +1,467 @@
+                TaggableFilter::forDefinition);
+        this.waterwayTagFilter = this.configurationValue(configuration, "waterway.tags.filters",
+                WATERWAY_TAG_FILTER_DEFAULT, TaggableFilter::forDefinition);
+
+        /* Ocean data */
+        this.validOceanTags = TaggableFilter.forDefinition(
+                this.configurationValue(configuration, "ocean.valid", DEFAULT_VALID_OCEAN_TAGS));
+        this.oceanBoundaryTags = TaggableFilter.forDefinition(this.configurationValue(configuration,
+                "ocean.boundary", DEFAULT_OCEAN_BOUNDARY_TAGS));
+
+        /* End ocean data */
+        /* Elevation settings */
+        this.minResolutionDistance = this.configurationValue(configuration,
+                "waterway.elevation.resolution.min.uphill", MIN_RESOLUTION_DISTANCE.asMeters(),
+                Distance::meters);
+        this.minDistanceStartEndElevationUphill = this.configurationValue(configuration,
+                "waterway.elevation.distance.min.start.end",
+                Distance.FIFTEEN_HUNDRED_FEET.asMeters(), Distance::meters);
+        /* End elevation settings */
+
+    }
+
+    /**
+     * Check if a line ends in an ocean
+     *
+     * @param line
+     *            The line to check
+     * @return {@code true} if the line ends in an ocean
+     */
+    public boolean doesLineEndInOcean(final LineItem line)
+    {
+        final Atlas atlas = line.getAtlas();
+        final PolyLine linePolyline = line.asPolyLine();
+        final Location last = linePolyline.last();
+        if (atlas.areasCovering(last, this.validOceanTags::test).iterator().hasNext()
+                || atlas.areasCovering(last, this.oceanBoundaryTags::test).iterator().hasNext())
+        {
+            return true;
+        }
+        final List<LineItem> lines = new ArrayList<>();
+        atlas.lineItemsIntersecting(line.asPolyLine().bounds(), this.oceanBoundaryTags::test)
+                .forEach(lines::add);
+        final LineItem[] intersecting = lines.stream()
+                .filter(l -> l.asPolyLine().intersects(linePolyline)).toArray(LineItem[]::new);
+        final SegmentIndexComparator segmentComparator = new SegmentIndexComparator(linePolyline);
+        for (final LineItem lineItem : intersecting)
+        {
+            final Collection<Pair<Segment, Segment>> segs = getIntersectingSegments(linePolyline,
+                    lineItem.asPolyLine());
+            final Segment max = segs.stream().map(Pair::getKey).distinct().max(segmentComparator)
+                    .orElse(null);
+            final Collection<Segment> crosses = segs.stream().filter(p -> p.getLeft().equals(max))
+                    .map(Pair::getValue).collect(Collectors.toSet());
+            final PolyLine coast = new PolyLine(crosses.stream()
+                    .flatMap(c -> Stream.of(c.first(), c.last())).toArray(Location[]::new));
+            if (isRightOf(coast, linePolyline.last())
```

Comment:
I am a bit confused what is happening at this point (some comments throughout this method would help). Is this considering a polyline to end in the ocean if it is to the east of a coast?

Reply:
There are two assumptions at play here:
1) `natural=coastline` follows the specification, and the ocean is to the right of the coastline way.
2) The geometry isn't so complicated that the waterway loops around the coastline (I can make a contrived example, but I haven't seen any real-world examples that couldn't be more accurately modeled).

I've gone ahead and added some documentation.

Reply:
I see, thanks for the clarification and the docs. It might be worth it to add a note about the ocean specification assumption in either the flag instructions or the MapRoulette challenge instructions in the configuration. That way editors may be able to identify incorrectly modeled coastlines that are getting caught by this check.